packages/block-editor/src/components/global-styles/typography-utils.js

```diff
@@ -185,3 +185,106 @@ export function findNearestFontWeight(
 
 	return nearestFontWeight;
 }
+
+/**
+ * Returns the nearest font style based on the new font style.
+ * Defaults to an empty string if the new font style is not valid or available.
+ *
+ * @param {Array}  availableFontStyles Array of available font weights.
+ * @param {string} newFontStyleValue   New font style value.
+ * @return {string} Nearest font style.
```

Comment:
Oh good spot! My intention was that it would default to an empty string if none of the conditionals were triggered. I've updated the function here: https://github.com/WordPress/gutenberg/pull/63215/commits/b407be1654b1ce6851b5a72e155180c39f27e9dd, I think this should cover all branches.